#Example 8.7
Calculate the height of a tower having its top in point A. The length of a line segment from B to C has been measured. Its prolongation goes through point D that is the foot point of the tower. Furthermore, the angles ABCD = $\alpha$ and ACD = $\beta$ have been measured. Calculate the height of the tower and estimate the error in the calculation. Wich of the measurements contributes most to the error in the result? Measurements: BC = $50 \pm 0.2$ m, $\alpha = (32.6 \pm 0.3)°$, $\beta = (53.8 \pm 0.3)°$.

The height of the tower can be calculated using the two angles $\alpha$ and $\beta$.
$$\begin{align*}
\tan \alpha & = \frac{|AD|}{|\overline{BC}|+|\overline{CD}|} \\
\tan \beta & = \frac{|\overline{AD}|}{|\overline{CD}|}
\end{align*}$$
Removing the unknown line segment $\overline{CD}$ we find
\begin{equation}
  h = h(|\overline{BC}|,\alpha,\beta) = |\overline{AD}| = \frac{\tan \alpha \tan \beta}{\tan \beta - \tan \alpha } |\overline{BC}|.
\end{equation}

In [1]:
from math import *

In [2]:
# Given measurements
BC=50.; E_bc=0.2;
alpha=32.6*pi/180.; E_alpha=0.3*pi/180.;
beta=53.8*pi/180.; E_beta=0.3*pi/180.;

## Height of the tower

In [3]:
# Calculate the height of the tower
def tower_height(BC, alpha, beta):
    return tan(alpha)*tan(beta)/(tan(beta)-tan(alpha))*BC 

In [4]:
h = tower_height(BC, alpha, beta); "h = |AD| = %.2f" % h

'h = |AD| = 60.11'

## Error estiatimation

We use partial derivatives of the measurement variables in order to estimate their contributions to the error in the result as well as the overall error.
$$
E_{AD} \approx \left | \frac{\partial h}{\partial |\overline{BC}|} \right | E_{BC} + \left | \frac{\partial h}{\partial \alpha} \right | E_\alpha + \left | \frac{\partial h}{\partial \beta} \right | E_\beta.
$$

$$\begin{align*}
  \frac{\partial h}{\partial |\overline{BC}|} & = \frac{\tan \alpha \tan \beta}{\tan \beta - \tan \alpha}
\end{align*}$$

$$\begin{align*}
\frac{\partial h}{\partial \alpha} & =
  |\overline{BC}| \frac{\frac{1}{\cos^2 \alpha}
                  \tan \beta \left (\tan \beta - \tan \alpha \right )
                        - \left (-\frac{1}{\cos^2 \alpha} \right ) \tan \alpha \tan \beta
                       }
                       {(\tan \beta - \tan \alpha)^2} \\
  & = |\overline{BC}| \frac{\tan^2 \beta}{\cos^2 \alpha (\tan \beta - \tan \alpha)^2}
\end{align*}$$

$$\begin{align*}
\frac{\partial h}{\partial \beta} & =
  |\overline{BC}| \frac{\frac{1}{\cos^2 \beta} \tan \alpha (\tan \beta - \tan \alpha)
                        - \frac{1}{\cos^2 \beta}(\tan \alpha \tan \beta)}
                       {(\tan \beta - \tan \alpha)^2} \\
  & = |\overline{BC}| \frac{-\tan^2 \alpha}{\cos^2 \beta(\tan \beta - \tan \alpha)^2}
\end{align*}$$

The contributions of the different measurement values to the error can be estimated as follows. From the measurement of the line segment between B and C we get
$E_{h,|\overline{BC}} \approx \left | \frac{\partial h}{\partial |\overline{BC}|} \right | E_{BC}$.
From the measurment of angle $\alpha$ we get
$E_{h,\alpha} \approx \left | \frac{\partial h}{\partial \alpha} \right | E_\alpha$.
From the measurement of angle $\beta$ we get
$E_{h,\beta} \approx \left | \frac{\partial h}{\partial \beta} \right | E_\beta$.

In [5]:
E_h_bc = abs(tan(alpha)*tan(beta)/(tan(beta)-tan(alpha)))*E_bc
E_h_alpha = BC*abs(tan(beta)**2/(cos(alpha)**2*(tan(beta)-tan(alpha))))*E_alpha
E_h_beta = BC*abs((-tan(alpha)**2)/(cos(beta)**2*(tan(beta)-tan(alpha))))*E_beta
print "E_h_bc    = %.2f m" % E_bc
print "E_h_alpha = %.2f m" % E_h_alpha
print "E_h_beta  = %.2f m" % E_h_beta

E_h_bc    = 0.20 m
E_h_alpha = 0.95 m
E_h_beta  = 0.42 m


In [6]:
E_h = E_h_bc + E_h_alpha + E_h_beta; "E_h = %.2f m" % E_h

'E_h = 1.61 m'

The limit of the absolute error in the height of the tower can be estimated as
$$E_h \approx E_{h,|\overline{BC}|}+E_{h,\alpha}+E_{h,\beta} = 1.61 \mathrm{m}.$$

## Experimental error estimation

In [7]:
E_h_bc = max(abs(tower_height(BC+E_bc, alpha, beta)-h),
             abs(tower_height(BC-E_bc, alpha, beta)-h))

In [8]:
E_h_alpha = max(abs(tower_height(BC, alpha+E_alpha, beta)-h),
                abs(tower_height(BC, alpha-E_alpha, beta)-h))

In [9]:
E_h_beta = max(abs(tower_height(BC, alpha, beta+E_beta)-h),
               abs(tower_height(BC, alpha, beta-E_beta)-h))

In [10]:
print "E_h_bc    = %.2f m" % E_bc
print "E_h_alpha = %.2f m" % E_h_alpha
print "E_h_beta  = %.2f m" % E_h_beta

E_h_bc    = 0.20 m
E_h_alpha = 1.32 m
E_h_beta  = 0.59 m


In [11]:
E_h = E_h_bc + E_h_alpha + E_h_beta; "E_h = %.2f m" % E_h

'E_h = 2.15 m'

The limit of the absolute error in the height of the tower can be estimated as
$$E_h \approx E_{h,|\overline{BC}|}+E_{h,\alpha}+E_{h,\beta} = 2.15 \mathrm{m}.$$